### PR TITLE
🌱  Add Dockerfiles for PubSub batch job

### PR DESCRIPTION
**Pull request by azeemshaikh38**
* **Please check if the PR fulfills these requirements**
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] PR title follows the guidelines defined in  https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)
Adds Dockerfiles for PubSub controller/worker. No behaviour change.

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)
No.

### CODE REVIEW

**Review by laurentsimon — APPROVED** [review]

File: cron/controller/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+COPY go.* ./
+RUN go mod download
+COPY . ./
+
+FROM base AS pubsub
+ARG TARGETOS
+ARG TARGETARCH
+RUN CGO_ENABLED=0 make build-pubsub
+
+FROM gcr.io/google.com/cloudsdktool/cloud-sdk:slim@sha256:acb34ca33bc83255110d16a749b0da618a96fcc481348ecfe614b3edcabdeaa6
```
Comment (naveensrinivasan) [inline cron/controller/Dockerfile:29]: Why do we need a `cloud-sdk:slim`? These are `go binaries` which does not use any of the cloudsdk tools like the command line. The previous cron used `gcloud command to upload to gcs`.

File: cron/worker/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+COPY go.* ./
+RUN go mod download
+COPY . ./
+
+FROM base AS pubsub
+ARG TARGETOS
+ARG TARGETARCH
+RUN CGO_ENABLED=0 make build-pubsub
+
+FROM gcr.io/google.com/cloudsdktool/cloud-sdk:slim@sha256:acb34ca33bc83255110d16a749b0da618a96fcc481348ecfe614b3edcabdeaa6
```
Comment (naveensrinivasan) [inline cron/worker/Dockerfile:29]: The same comment applies as above.

File: cron/controller/Dockerfile (hunk 1 above)
Comment (azeemshaikh38) [reply cron/controller/Dockerfile:29]: Ah. I just copied this as is from `cron/Dockerfile`. So does this only need a single `FROM` which would be the `golang:1.16.4@sha` import?

Comment (naveensrinivasan) [reply cron/controller/Dockerfile:29]: It has to be similar to this one https://github.com/ossf/scorecard/blob/main/Dockerfile#L28. It is from distroless container.

Comment (azeemshaikh38) [reply cron/controller/Dockerfile:29]: Thanks! Working on a PR, will send it out shortly.

**Pull request merged**